core/src/stored/checkpointhandler.h

```diff
@@ -0,0 +1,49 @@
+/*
+   BAREOS® - Backup Archiving REcovery Open Sourced
+
+Copyright (C) 2023-2023 Bareos GmbH & Co. KG
+
+This program is Free Software; you can redistribute it and/or
+modify it under the terms of version three of the GNU Affero General Public
+License as published by the Free Software Foundation and included
+in the file LICENSE.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+02110-1301, USA.
+*/
+
+#ifndef BAREOS_STORED_CHECKPOINTHANDLER_H_
+#define BAREOS_STORED_CHECKPOINTHANDLER_H_
+
+#include "include/jcr.h"
+
+namespace storagedaemon {
+
+class CheckpointHandler {
+ public:
+  CheckpointHandler(time_t interval);
+  void DoBackupCheckpoint(JobControlRecord* jcr);
+  void DoTimedCheckpoint(JobControlRecord* jcr);
+  void DoVolumeChangeBackupCheckpoint(JobControlRecord* jcr);
+  void SetReadyForCheckpoint(bool ready) { ready_for_checkpoint_ = ready; }
+  bool ReadyForCheckpoint() const { return ready_for_checkpoint_; }
```

Comment:
While this looks completely reasonable here in context, `ReadyForCheckpoint()` could also mean "I am ready for a checkpoint now" instead of "Am I ready for a checkpoint?"
Mark at least as `[[nodiscard]]` or rename to `IsXYZ()`.